### PR TITLE
Preparation for v0.19

### PR DESCRIPTION
**Pull request by GearsAD**
This is in preparation for the v0.19 release as we discussed @dehann @Affie.

- Attic has been cleaned out
- Neo4jDFG has been removed
- Tests have been updated

Open question(s) (@Affie)
- Is GraphsDFG different from LightDFG? Can we not merge those?

Regarding downstream the DFG driver we are planning on building out, I believe these functions will be difficult to implement:
- getAddHistory
- getBiadjacencyMatrix
- getInternalId
- findFactorsBetweenNaive
- saveDFG, loadDFG!, loadDFG
- toDot and toDotFile

I suggest we leave this open until we are ready to make v0.19 (no urgency), I just set it up while reviewing the code.

### CODE REVIEW

**Comment by dehann** [comment]
>    getAddHistory
>    getBiadjacencyMatrix
>    getInternalId
>    findFactorsBetweenNaive
>    saveDFG, loadDFG!, loadDFG
>    toDot and toDotFile

All these functions should stay in the abstract interface, and if new drivers are not immediately able to implement them then that is just a limitation of that driver.  Other DFG drivers like GraphDFG already implement them so nothing has to change there.  Also functions that exist in AbstactDFG only made it there because they were used somewhere.  We have very little fat in DFG so to speak.

**Comment by dehann** [comment]
> to make v0.19 (no urgency), I just set it up while reviewing the code

thanks for prepping the PR!

**Comment by dehann** [comment]
> Neo4jDFG has been removed

We should just put his under a whole new repo and distributed with GPL license?  See Stefan K's issue:
- https://github.com/JuliaRobotics/KernelDensityEstimate.jl/issues/94

File: src/services/Serialization.jl [hunk 1]
```diff
@@ -232,8 +232,15 @@ function unpackVariable(
         Dict{Symbol, MeanMaxPPE}()
     end
 
-    smallData = JSON2.read(packedProps["smallData"], Dict{Symbol, SmallDataTypes})
-
+    # Smalldata refactor to metadata
+    if (haskey(packedProps, "metadata"))
```
Comment (dehann) [inline src/services/Serialization.jl:236]: Just to help, on master branch i added additional option in if packedProps is no longer a JSON string inside the top level JSONable object:
https://github.com/JuliaRobotics/DistributedFactorGraphs.jl/blob/685b5f99a4a1c7a933bcf8dd9af81ecb759c0a7a/src/services/Serialization.jl#L348-L360

**Review by dehann — COMMENTED** [review]
looks good to me

**Comment by GearsAD** [comment]
This is the precursor to resolving #590

**Comment by GearsAD** [comment]
-5,700 lines... yep :broom: :fire:

**Review by dehann — APPROVED** [review]
i had another fix or two to add, and looks good to me.  Tests on IIF also pass.

**Pull request merged**